packages/http/src/route.ts

```diff
@@ -215,6 +215,17 @@ export function setRoute(context: DecoratorContext, entity: Type, details: Route
 
   const state = context.program.stateMap(routesKey);
 
+  // verify that all shared routes agree on the value of the shared flag
+  const existing = [...state.values()].filter((value) => value.path === details.path);
```

Comment:
Should this maybe move to `onValidate` instead of being checked every time a new route is added?

Reply:
that would also then make it easier to report the diagnostic on each routes without duplicates using the `DuplicateTracker`

Reply:
Relocated the logic to `onValidate`.